src/metabase/util/urls.clj

```diff
@@ -5,47 +5,60 @@
 
    Functions for generating URLs not related to Metabase *objects* generally do not belong here, unless they are used in many places in the
    codebase; one-off URL-generation functions should go in the same namespaces or modules where they are used."
-  (:require [metabase.public-settings :as public-settings]))
+  (:require [metabase.plugins.classloader :as classloader]
+            [metabase.public-settings :as public-settings]
+            [metabase.util :as u]))
+
+;; we want to load this at the top level so the Setting the namespace defines gets loaded
+(def ^:private site-url*
+  (or (u/ignore-exceptions
+        (classloader/require 'metabase-enterprise.advanced-config.util.urls)
+        (resolve 'metabase-enterprise.advanced-config.util.urls/site-url))
+      (constantly nil)))
+
+(defn site-url
```

Comment:
```suggestion
(defn- site-url
```
This can be made private. Will also get rid of the linter error complaining about a missing docstring (although it wouldn't hurt to add a docstring as well)

Reply:
We didn't get rid of the linter that checks for missing docstrings. Clj-Kondo checks for them

Reply:
Oops sorry I misread that comment. Nevermind